src/tool_parsecfg.c

```diff
@@ -135,13 +146,18 @@ int parseconfig(const char *filename, struct GlobalConfig *global)
 
   if(file) {
     char *line;
-    char *aline;
     char *option;
     char *param;
     int lineno = 0;
     bool dashed_option;
+    char *aline = malloc(MAX_CONFIG_LINE_LENGTH);
+    if(!aline) {
+      if(file != stdin)
+        fclose(file);
+      return 2;
```

Comment:
We return 1 elsewhere for no memory. I'm not sure why you chose 2 so I left it as is.